tutorials/scripting/gdextension/what_is_gdextension.rst

```diff
@@ -0,0 +1,99 @@
+
+Unlike modules, GDExtension doesn't require compiling the engine's source code,
+making it easier to distribute your work. It gives you access to most of the API
+available to GDScript C#, allowing you to code game logic with full control
+regarding performance. It's ideal if you need high-performance code you'd like
+to distribute as an add-on in the :ref:`asset library <doc_what_is_assetlib>`.
+
+Also:
+
+- GDExtension is not limited to C and C++. Thanks to :ref:`third-party bindings
+  <doc_what_is_gdextension_third_party_bindings>`, you can use it with many other
+  languages.
+- You can use the same compiled GDExtension library in the editor and exported
+  project. With C++ modules, you have to recompile all the export templates you
+  plan to use if you require its functionality at run-time.
+- GDExtension only requires you to compile your library, not the whole engine.
+  That's unlike C++ modules, which are statically compiled into the engine.
+  Every time you change a module, you need to recompile the engine. Even with
+  incremental builds, this process is slower than using GDExtension.
+
+Advantages of C++ modules
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We recommend :ref:`C++ modules <doc_custom_modules_in_c++>` in cases where
+GDExtension isn't enough:
+
+- C++ modules provide deeper integration into the engine. GDExtension's access is not as deep as
+  static modules
+- You can use C++ modules to provide additional features in a project without
+  carrying native library files around. This extends to exported projects.
+(- C++ modules can be faster than GDExtension, especially when the code requires a
+  lot of communication through the scripting API.) -> TODO: check if this still holds up
+
+Supported languages
+-------------------
+
+The Godot developers officially support the following language bindings for
+GDExtension:
+
+- C++ :ref:`(tutorial) <doc_gdextension_cpp_example>`
+
+.. note::
+
+    There are no plans to support additional languages with GDExtension officially.
+    That said, the community offers several bindings for other languages (see
+    below).
+
+.. _doc_what_is_gdextension_third_party_bindings:
+
+The bindings below are developed and maintained by the community:
+
+.. Binding developers: Feel free to open a pull request to add your binding if it's well-developed enough to be used in a project.
+.. Please keep languages sorted in alphabetical order.
+
+- `Rust <https://github.com/godot-rust/godot-rust>`__
+
+.. note::
+
+    Not all bindings mentioned here may be production-ready. Make sure to
+    research options thoroughly before starting a project with one of those.
+    Also, double-check whether the binding is compatible with the Godot version
+    you're using.
+
+Version compatibility
+---------------------
+
+(:ref:`Unlike Godot itself <doc_release_policy>`, GDExtension has stricter version
+compatibility requirements as it relies on low-level *ptrcalls* to function.) -> TODO: check up on this
```

Comment:
Yeah, I am not sure if this is correct. I have not enough knowledge about the C++ base system to know if that is correct

Reply:
Time will tell how well we implemented things but GDExtensions were created as resilient as possible. 
As long as we add to the API in Godot and not remove functions, GDExtensions targeting older versions of Godot will likely work on newer versions. But it is very likely extensions will have to be altered and recompiled when going from say Godot 4.0 to 4.1.

That said, no guarantees here as it's easy to do something in Godot that will break extensions without any checks in place to prevent us from doing so.